android/app/src/main/java/org/fossasia/openevent/viewmodels/ScheduleFragmentViewModel.java

```diff
@@ -0,0 +1,51 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ScheduleFragmentViewModel extends ViewModel {
+
+    private RealmDataRepository realmRepo;
+
+    public ScheduleFragmentViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<List<String>> getEventDatesString() {
+        MutableLiveData<List<String>> eventDatesStringLiveData = new MutableLiveData<>();
+        List<String> eventDatesString = new ArrayList<>();
+        realmRepo.getEventDates()
+                .addChangeListener((eventDates, orderedCollectionChangeSet) -> {
+                    eventDatesString.clear();
+                    int eventDays = eventDates.size();
+                    for (int i = 0; i < eventDays; i++) {
+                        String date = eventDates.get(i).getDate();
+                        eventDatesString.add(date);
+                    }
+                    eventDatesStringLiveData.setValue(eventDatesString);
+        });
+        return eventDatesStringLiveData;
+    }
+
+    public LiveData<List<Track>> getTracks() {
+        MutableLiveData<List<Track>> tracksLiveData = new MutableLiveData<>();
+        realmRepo.getTracks().addChangeListener((tracks, orderedCollectionChangeSet) -> {
+            tracksLiveData.setValue(tracks);
+        });
```

Comment:
I didn't realize this in other PRs but this approach is wrong. this attaches the listener each time and also creates a new live data as well, the live data and listener should be created once and stored as a reference in the ViewModel and just returned to the view in each call. The onCleared method should be clearing the stored reference instead of getting a new reference and cleared. Please fix this in this PR and also raise issues for previous one as well